libs/community/langchain_community/chat_message_histories/cassandra.py

```diff
@@ -19,27 +22,37 @@
 DEFAULT_TTL_SECONDS = None
 
 
-class CassandraChatMessageHistory(BaseChatMessageHistory):
-    """Chat message history that stores history in Cassandra.
-
-    Args:
-        session_id: arbitrary key that is used to store the messages
-            of a single chat session.
-        session: Cassandra driver session. If not provided, it is resolved from cassio.
-        keyspace: Cassandra key space. If not provided, it is resolved from cassio.
-        table_name: name of the table to use.
-        ttl_seconds: time-to-live (seconds) for automatic expiration
-            of stored entries. None (default) for no expiration.
-    """
+def _rows_to_messages(rows: Iterable[RowType]) -> List[BaseMessage]:
+    message_blobs = [row["body_blob"] for row in rows][::-1]
+    items = [json.loads(message_blob) for message_blob in message_blobs]
+    messages = messages_from_dict(items)
+    return messages
+
 
+class CassandraChatMessageHistory(BaseChatMessageHistory):
     def __init__(
         self,
         session_id: str,
         session: Optional[Session] = None,
         keyspace: Optional[str] = None,
         table_name: str = DEFAULT_TABLE_NAME,
         ttl_seconds: Optional[int] = DEFAULT_TTL_SECONDS,
+        *,
+        setup_mode: Optional[SetupMode] = SetupMode.SYNC,
```

Comment:
What's your perspective on `setup_mode` vs. having a dedicated `async_session` ?

Reply:
The cassandra `Session` is used for both sync and async so there's no "async session"